lib/dom/xml_entities.nit

```diff
@@ -0,0 +1,586 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# This file is free software, which comes along with NIT.  This software is
+# distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without  even  the implied warranty of  MERCHANTABILITY or  FITNESS FOR A
+# PARTICULAR PURPOSE.  You can modify it is you want,  provided this header
+# is kept unaltered, and a notification of the changes is added.
+# You  are  allowed  to  redistribute it and sell it, alone or is a part of
+# another product.
+
+# Basic blocks for XML parsing
+#
+# DOM entities are defined in this module, specifically:
+#
+# * XMLEntity: Abstract kind of XML-related node
+# * XMLDocument: A well-formed XML document, root of the tree
+# * PCDATA: Raw XML-escaped character data
+# * CDATA: Raw data, may contain invalid XML escape characters
+# * XMLTag: Abstract XML tag element
+# * XMLAttrTag: Abstract XML element, they may contain attributes
+# * XMLOnelinerTag: Any tag contained on one-line only
+# * XMLStartTag: A tag starting a new hierarchy level in the tree
+# * XMLPrologTag: A tag containing meta-information on the document
+# * XMLEndTag: A tag signaling the end of a block
+# * XMLCommentTag: A comment tag
+# * XMLSpecialTag: A special tag, which may contain meta-information (DOCTYPE for instance)
+# * XMLAttribute: Any kind of attribute that may be attached to a tag
+# * XMLStringAttr: An attribute containing a String
+# * XMLError: Any kind of error thrown while parsing a document
+module xml_entities
+
+# Information about the location of an entity in an XML source document
+private class Location
+	# Line in which the element is described
+	var line: Int
+	# Offset in the line at which the element is positioned
+	var offset: Int
+
+	redef fun to_s do return "line {line}, position {offset}"
+end
+
+# Any kind of XML Entity
+abstract class XMLEntity
+	# Optional parent of `self`
+	var parent: nullable XMLEntity is private writable(set_parent)
+
+	# Optional location of the entity in source
+	private var location: nullable Location
+
+	# The children of `self`
+	var children = new HashSet[XMLEntity]
+
+	# Sets the parent of `self` to `e`
+	fun parent=(e: XMLEntity) do
```

Comment:
Is a parent setter an usual thing in a XML DOM API (so much acronyms)? I my non-ex(ml)pert mind I think that the responsibility to attach the parent to a children should be in the parent class as it can decides where to put the children (at the begin, at the end, in replacement of other elements, etc) and, why not, offers multiple services